hazelcast/src/main/java/com/hazelcast/util/ExponentialBackoffSleeper.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.util;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Utility class that allows for polling of state, with an exponentially increasing delay. The
+ * {@link Builder#factor exponential factor} can be specified to control how fast the wait time increases between
+ * subsequent invocations of {@link #sleep}.
+ * <p>
+ * Optionally, a {@link Builder#timeoutMs timeout} can be specified to limit the sleep time to a global timeout.
+ */
+public final class ExponentialBackoffSleeper {
+    /**
+     * Default exponential factor
+     */
+    public static final double DEFAULT_FACTOR = 1.3d;
```

Comment:
You always set the factor to the same value as the default. Is it an attempt to be explicit? Couldn't we simplify  building instances of this class?